File: pset6/readability/readability.py
from cs50 import get_string
import re


def main():
    word = get_string("Text:")
    listWord = numberWords(word)
    Sentences = numberSentence(word)
    Words = len(listWord)
    Letters = letterCount(word)
    L = (Letters/Words)*100
    S = (Sentences/Words)*100
    grade = calculate(L, S)
    
    if (0 < grade <= 16):
        print("Grade:", grade)
    elif(grade < 1):
        print("Before Grade 1")
    elif(grade > 16):
        print("Grade 16+")
    
    # print(Words)
    # print(Letters)
    # print(Sentences)
    # print(grade)


def numberWords(word):
    words = list(map(len, word.split()))
    return words
    

def numberSentence(word):
    split = re.split('(?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?|\!)\s', word)
    S = len(split)
    return S
    

def letterCount(L):
    letters = sum(c.isalpha() for c in L)
    return letters
    

def calculate(L, S):
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    # print("grade::: ", grade)
    return grade


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");
    float letters = 0;
    float words = 1;
    float sentence = 0;

    // convert all chart to upper
    for (int i = 0; i < strlen(s); i++)
    {
        char upperChar = toupper(s[i]);
        if (upperChar >= 'A' && upperChar <= 'Z')
        {
            letters += 1;
        }
    }

    // calcualte lenght of each word
    for (int j = 0; j < strlen(s); j++)
    {
        if (s[j] == ' ' && (s[j - 1] != ' ' && s[j + 1] != ' '))
        {
            words += 1;
        }
    }

    // calculate lenght of each sentence
    for (int k = 0; k < strlen(s); k++)
    {
        if (s[k] == '.' || s[k] == '?' || s[k] == '!')
        {
            sentence += 1;
        }
    }
    // use formula to caluclate grade read level
    float L = (letters / words) * 100;
    float S = (sentence / words) * 100;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    // printf("letter count: %f\n", letters);
    // printf("words count: %f\n", words);
    // printf("sentence count: %f\n", sentence);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}